gst/nnstreamer/tensor_filter/tensor_filter.c

```diff
@@ -429,6 +454,45 @@ record_statistics (GstTensorFilterPrivate * priv)
   }
 }
 
+
+/**
+ * @brief Track estimated latency and notify pipeline when it changes.
+ *        Latency estimates may be a bit jittery. On the principle we want to
+ *        inform pipeline with the latency from longest inference.
+ *        However, first inference may take much longer, or model filter
+ *        configuration may change. Therefore any change of more than 10%
+ *        (arbitrary value) to a lower latency is also reported to pipeline.
+ *        Notification is done sending LATENCY message to bus. Upon receipt,
+ *        application will initiate a pipeline latency probe via LATENCY query.
+ */
+static void
+track_latency (GstTensorFilter * self)
+{
+  GstTensorFilterPrivate *priv = &self->priv;
+  gdouble estimated, reported, deviation;
+
```

Comment:
How about check 'priv->latency_reporting' in here?
if (!priv->latency_reporting)
  return;
or remove line 479 "(priv->latency_reporting) && " and add it to line 789

479: if (estimated > 0) {

789: if (priv->latency_reporting)
           track_latency (self);

Reply:
Agreed, such slightly different logic could be used with similar functional behavior.
Change has been merged already so can not integrate in this PR - suggestion could make it into future change.
Thanks